pkgs/applications/networking/browsers/firefox/env_var_for_system_dir-ff86.patch

```diff
@@ -0,0 +1,21 @@
+diff -r 22fc47c968f2 toolkit/xre/nsXREDirProvider.cpp
+--- a/toolkit/xre/nsXREDirProvider.cpp	Mon Dec 14 15:09:17 2020 +0000
++++ b/toolkit/xre/nsXREDirProvider.cpp	Tue Feb 23 23:38:56 2021 +0100
+@@ -11,6 +11,7 @@
+ 
+ #include "jsapi.h"
+ #include "xpcpublic.h"
++#include "prenv.h"
+ 
+ #include "nsIAppStartup.h"
+ #include "nsIFile.h"
+@@ -305,7 +306,8 @@
+       "/usr/lib/mozilla"_ns
+ #    endif
+       ;
+-  rv = NS_NewNativeLocalFile(dirname, false, getter_AddRefs(localDir));
++  const char* pathVar = PR_GetEnv("MOZ_SYSTEM_DIR");
```

Comment:
Would using `getenv()` instead of `PR_GetEnv()` in the original patch work? This way there could be only one patch. From a quick glance over `nsXREDirProvider.cpp` seems to already have `getenv()` so it should work for both the ESR and regular variants of Firefox.

Reply:
Possibly, but the price of updating firefox is already high enough that I opt for convenience in this case. FWIW: I tried applying getenv over PR_GetEnv but the patch wouldn't immediately apply on FF78, so I scrapped the idea.